src/main/mixpanel.ts

```diff
@@ -0,0 +1,48 @@
+import { v4 as ipv4 } from "public-ip";
+import { Mixpanel } from "mixpanel";
+
+export interface INineChroniclesMixpanel {
+    track(eventName: string, properties?: object, callback?: () => void): void;
+    alias(alias: string): void;
+    login(): void;
+    miningConfig(): void;
+}
+
+export class NineChroniclesMixpanel implements INineChroniclesMixpanel {
+    private readonly _mixpanel: Mixpanel;
+    private readonly _mixpanelUuid: string;
+    private readonly _ip: string;
+    private _login: boolean = false;
+    private _miningConfig: boolean = false;
+
+    constructor(mixpanel: Mixpanel, mixpanelUuid: string) {
+        this._mixpanel = mixpanel;
+        this._mixpanelUuid = mixpanelUuid;
+        let ip: string = 'unknown'
+        ipv4().then((value) => (ip = value));
+        this._ip = ip;
```

Comment:
```suggestion
        ipv4().then((value) => (this._ip = value));
```

.then() is called as a microtask, which means the handler will run **after** the constructor call. This will result in `_ip` being always unknown.

Reply:
But `this._ip` is `readonly` property. so we cannot init here. Have you a good idea about this?

Reply:
For now just keep it mutable would be reasonable enough. It was already managed as mutable value before, so I think it is a reasonable compromise.

Reply:
Hmm... ok, I'll make this variable as a mutable. But I guess we need some comments here like 'This variable must not be changed.` to prevent mistakes in other engineers.

Reply:
apply on f074d11